code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -2239,3 +2239,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
 /datum/reagent/consumable/ethanol/pruno/on_mob_life(mob/living/carbon/M)
 	M.adjust_disgust(5)
 	..()
+
+/datum/reagent/consumable/ethanol/brass_monkey
+	name = "Brass Monkey"
+	color = "#e9e357"
+	description = "An ancient beverage supposedly invented by a bunch of monkeys trying to join a failing steampunk cult. Traditionally served in a clockwork mug."
+	boozepwr = 85
+	taste_description = "steam-powered bananas"
+	glass_icon_state = "brass_monkey"
+	glass_name = "Brass Monkey"
+	glass_desc = "Coolin' by the lockers, getting kind of funky, me and the crew, we're drinking Brass Monkey."
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_metabolize(mob/living/M)
+	if(!HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
+		ADD_TRAIT(M, TRAIT_MONKEYBRO, BRASS_MONKEY)
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_end_metabolize(mob/living/M)
+	if(HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
```

Comment:
ditto

Reply:
You're probably right, but I want to go on the safe side of things. Besides, I seriously doubt one HAS_TRAIT_FROM proc whenever someone sips this drink will have a noticeable effect on performance.

Reply:
that's not a good argument. on its own barely any check will really dent performance on its own.  It's the accumulative dent tons of those checks would have if noone cared that would affect performance, though.

@Rohesie you worked with the trait code, does it matter if you remove / add traits without checking first? I'd imagine not right?

Reply:
This is very much a no.
This check is superfluous at best and error-hiding at worst.
Use `ADD_TRAIT(M, TRAIT_MONKEYBRO, type)` and `REMOVE_TRAIT(M, TRAIT_MONKEYBRO, type)`
No need to define a new string for a reagent, the type is the convention.